src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter(person => person.sex === 'm');
+
+  if (century) {
+    men = men.filter(man => Math.ceil(man.died / 100) === century);
+  }
+
+  const ages = men.map(man => man.died - man.born);
+  const sumOfAges = ages.reduce((prev, x) => prev + x, 0);
+
+  return sumOfAges / ages.length;
```

Comment:
create a callback function for all these calculations, you repeat the same logic three times